webapp/src/ts/components/snackbar/snackbar.component.ts

```diff
@@ -39,39 +42,59 @@ export class SnackbarComponent implements OnInit {
 
   ngOnInit() {
     this.changeDetectorRef.detach();
-    const reduxSubscription = this.store.select(Selectors.getSnackbarContent).subscribe(content => {
-      if (!content) {
+    const reduxSubscription = this.store.select(Selectors.getSnackbarContent).subscribe((snackbarContent) => {
+      if (!snackbarContent?.message) {
+        this.message = undefined;
+        this.action = undefined;
+        this.hide();
+
         return;
       }
 
+      const { message, action } = snackbarContent;
       if (this.active) {
-        this.hide(false);
-        this.setTimeout(() => this.show(content), this.ANIMATION_DURATION);
+        clearTimeout(this.resetMessageTimeout);
+        this.resetMessageTimeout = this.setTimeout(() => this.resetMessage(), this.ANIMATION_DURATION);
+        this.queueShowMessage(message, action);
```

Comment:
This is the only place where this is called right? Can we move clearing the timeout and setting the timeout to this function?